pkg/pex/store.go

```diff
@@ -210,7 +215,7 @@ func randomOrder() query.OrderByFunction {
 
 func filter(f func(*GossipRecord) bool) func(gossipSlice) gossipSlice {
 	return func(gs gossipSlice) gossipSlice {
-		filtered := gs[:0]
+		filtered := make(gossipSlice, 0)
```

Comment:
Let's do `make(gossipSlice, 0, len(gs))`.

It surely doesn't matter in practice, but since we know the final length of the slice, we may as well avoid any extra allocs.

Reply:
We don't know the final length, as we are filtering the elements of `gs`. So the final length may be smaller.

Reply:
Yeah, I misspoke:  we know the maximum size, which is exactly `len(gs)`.  The expression `make(gossipSlice, 0, len(gs))` will give us a zero-length slice, but whose underlying array has the max size, thus avoiding allocs.

Again, this probably doesn't matter for our current use, but it doesn't hurt.